pkg/asset/ignition/bootstrap/gcp/storage.go

```diff
@@ -0,0 +1,131 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	gcpic "github.com/openshift/installer/pkg/asset/installconfig/gcp"
+)
+
+const (
+	bootstrapIgnitionBucketObjName = "bootstrap.ign"
+)
+
+// GetBootstrapStorageName gets the name of the storage bucket for the bootstrap process.
+func GetBootstrapStorageName(clusterID string) string {
+	return fmt.Sprintf("%s-bootstrap-ignition", clusterID)
+}
+
+// NewStorageClient creates a new Google storage client.
+func NewStorageClient(ctx context.Context) (*storage.Client, error) {
+	ssn, err := gcpic.GetSession(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get session while creating gcp storage client: %w", err)
+	}
+
+	client, err := storage.NewClient(ctx, option.WithCredentials(ssn.Credentials))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %w", err)
+	}
+
+	return client, nil
+}
+
+// CreateBucketHandle will create the bucket handle that can be used as a reference for other storage resources.
+func CreateBucketHandle(ctx context.Context, bucketName string) (*storage.BucketHandle, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*1)
+	defer cancel()
+
+	client, err := NewStorageClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create storage client: %w", err)
+	}
+	return client.Bucket(bucketName), nil
+}
+
+// CreateStorage creates the gcp bucket/storage. The storage bucket does Not include the bucket object. The
+// bucket object is created as a separate process/function, so that the two are not tied together, and
+// the data stored inside the object can be set at a later time.
+func CreateStorage(ctx context.Context, ic *installconfig.InstallConfig, bucketHandle *storage.BucketHandle, clusterID string) error {
+	labels := map[string]string{}
+	labels[fmt.Sprintf("kubernetes-io-cluster-%s", clusterID)] = "owned"
+	for _, label := range ic.Config.GCP.UserLabels {
+		labels[label.Key] = label.Value
+	}
+
+	bucketAttrs := storage.BucketAttrs{
+		UniformBucketLevelAccess: storage.UniformBucketLevelAccess{
+			Enabled: true,
+		},
+		Location: ic.Config.GCP.Region,
+		Labels:   labels,
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*1)
+	defer cancel()
+
+	if err := bucketHandle.Create(ctx, ic.Config.GCP.ProjectID, &bucketAttrs); err != nil {
+		return fmt.Errorf("failed to create bucket: %w", err)
+	}
+	return nil
+}
+
+// CreateSignedURL creates a signed url and correlates the signed url with a storage bucket.
+func CreateSignedURL(handle *storage.BucketHandle, objectName string) (string, error) {
+	// Signing a URL requires credentials authorized to sign a URL. You can pass
```

Comment:
Signblob permission is only needed if the authentication doesn't have a key included right?